lib/posix/clock.c

```diff
@@ -18,14 +18,30 @@
  */
 static struct timespec rt_clock_base;
 
+#ifdef POSIX_CLOCK_MICROSECONDS
+#define UPTIME_PER_SEC USEC_PER_SEC
+#define NSEC_PER_UPTIME	(NSEC_PER_USEC)
+static inline int64_t uptime_get(void)
+{
+	return k_uptime_get_us();
+}
+#else
+#define UPTIME_PER_SEC MSEC_PER_SEC
+#define NSEC_PER_UPTIME	(USEC_PER_MSEC * NSEC_PER_USEC)
+static inline int64_t uptime_get(void)
+{
+	return k_uptime_get();
+}
+#endif
```

Comment:
This seems needlessly complicated.  Just get the time in ticks unconditionally and convert using k_ticks_to_ns_floor64().  You'll always be as close to ns precision as the tick rate allows, no fiddling with kconfig needed.